### ISSUE TITLE
Possible Compiler Regression in 1.7.x

### ISSUE THREAD

**Issue by toddsundsted** [crystal-lang/crystal#12974]
## Bug Report

I didn't see this called out in the changelog or previous issues as an intentional change, so I'm raising it as a possible compiler regression.

The following code compiles in versions of the compiler up to 1.6.2:
```
module Foo end
class Bar include Foo end
```

It no longer compiles in 1.7.x. The following does compile, as expected:
```
module Foo end
class Bar
  include Foo
end
```

**Comment by Blacksmoke16** [comment]
Caused via https://github.com/crystal-lang/crystal/pull/12622. Which was listed as a breaking change, but still might be a regression in this particular case.

Also I didn't realize you could have inlined declarations like that without using `;`.

**Comment by asterite** [comment]
I think we should fix it. It works in Ruby just fine and I see no reason to disallow such syntax.

**Comment by straight-shoota** [comment]
Yeah, that breaking is not acceptable.

**Comment by Sija** [comment]
IMO such a breaking change is better than the syntax with even more exceptions.

**Comment by asterite** [comment]
The thing is that there were no exceptions before. And none was needed. `class Foo include Bar end` was fine. `class Foo:bar end` also kind of works: it's a class name followed by a symbol. It works in Ruby too. It doesn't look good, but it makes sense from a lexer/parser perspective. We wanted to fix that (the `Foo:bar` thing which looks strange) and with it we broke something else.

**Comment by straight-shoota** [comment]
@Sija If it breaks working code, it's never better. In *some* cases this can be unavoidable to fix serious bugs.

But in this case, it's completely avoidable.

That doesn't mean that the changes from #12622 are undesired. But they're not necessary. We can postpone them tinker with it and decide what syntax should be accepted and what not. We were expecting this to only affect obscure edge cases that are never used. No big deal. But apparently it affects real code, and that we should prevent.

**Comment by toddsundsted** [comment]
thank you, all!

**Comment by straight-shoota** [comment]
Resolved by #12977
